core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/entity/EntitySelectorFactory.java

```diff
@@ -198,71 +199,65 @@ private EntitySelector<Solution_> applyFiltering(EntitySelector<Solution_> entit
-                    + ") has both a sorterManner (" + config.getSorterManner()
-                    + ") and a sorterWeightFactoryClass (" + config.getSorterWeightFactoryClass() + ").");
-        }
-        if (config.getSorterManner() != null && config.getSorterClass() != null) {
-            throw new IllegalArgumentException("The entitySelectorConfig (" + config
-                    + ") has both a sorterManner (" + config.getSorterManner()
-                    + ") and a sorterClass (" + config.getSorterClass() + ").");
-        }
-        if (config.getSorterManner() != null && config.getSorterOrder() != null) {
-            throw new IllegalArgumentException("The entitySelectorConfig (" + config
-                    + ") with sorterManner (" + config.getSorterManner()
-                    + ") has a non-null sorterOrder (" + config.getSorterOrder() + ").");
+            throw new IllegalArgumentException("""
+                    The entitySelectorConfig (%s) with sorterManner (%s) \
+                    and sorterComparatorClass (%s) and sorterWeightFactoryClass (%s) and sorterOrder (%s) and sorterClass (%s) \
+                    has a resolvedSelectionOrder (%s) that is not %s."""
+                    .formatted(config, sorterManner, config.getSorterComparatorClass(),
+                            config.getSorterWeightFactoryClass(), config.getSorterOrder(), config.getSorterClass(),
+                            resolvedSelectionOrder, SelectionOrder.SORTED));
         }
+        assertNotSorterMannerAnd(config, "sorterComparatorClass", EntitySelectorConfig::getSorterComparatorClass);
+        assertNotSorterMannerAnd(config, "sorterWeightFactoryClass", EntitySelectorConfig::getSorterWeightFactoryClass);
+        assertNotSorterMannerAnd(config, "sorterClass", EntitySelectorConfig::getSorterClass);
+        assertNotSorterMannerAnd(config, "sorterOrder", EntitySelectorConfig::getSorterOrder);
+        assertNotSorterClassAnd(config, "sorterComparatorClass", EntitySelectorConfig::getSorterComparatorClass);
+        assertNotSorterClassAnd(config, "sorterWeightFactoryClass", EntitySelectorConfig::getSorterWeightFactoryClass);
+        assertNotSorterClassAnd(config, "sorterOrder", EntitySelectorConfig::getSorterOrder);
         if (config.getSorterComparatorClass() != null && config.getSorterWeightFactoryClass() != null) {
-            throw new IllegalArgumentException("The entitySelectorConfig (" + config
-                    + ") has both a sorterComparatorClass (" + config.getSorterComparatorClass()
-                    + ") and a sorterWeightFactoryClass (" + config.getSorterWeightFactoryClass() + ").");
+            throw new IllegalArgumentException(
+                    "The entitySelectorConfig (%s) has both a sorterComparatorClass (%s) and a sorterWeightFactoryClass (%s)."
+                            .formatted(config, config.getSorterComparatorClass(), config.getSorterWeightFactoryClass()));
```

Comment:
Let's create a variable for `config.getSorterComparatorClass()` and `config.getSorterWeightFactoryClass()`.